edgelet/edgelet-test-utils/src/module.rs

```diff
@@ -11,37 +11,38 @@ use futures::stream;
 use futures::IntoFuture;
 use hyper::Body;
 
-#[derive(Clone, Debug)]
-pub struct NullRegistry<E: Fail> {
-    phantom: PhantomData<E>,
+#[derive(Clone, Debug, Default)]
```

Comment:
`#[derive(Default)` will require `E: Default` even though it isn't required (`Option<E>` is always `Default`). If you really want this to impl `Default` as an alternative to `TestRegistry::new(None)`, then you probably want to impl `Default` manually.

```rust
impl<E> Default for TestRegistry<E> {
    fn default() -> Self {
        TestRegistry {
            err: None,
        }
    }
}
```

Reply:
Removed Default as it is not necessary

---
In reply to: [243432315](https://github.com/Azure/iotedge/pull/656#discussion_r243432315) [](ancestors = 243432315)